CannibalisticZombies/Assets/Scripts/Inventory.cs

```diff
@@ -0,0 +1,73 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class Inventory 
+{
+    private List<InventorySlot> inventory;
+    private int maxWeight;
+
+    public Inventory() 
+    {
+        inventory = new List<InventorySlot>();
+    }
+
+    //-/////////////////////////////////////////////////////////////////////
+    /// adds amount of new item to existing InventorySlot with said item
+    /// if maxWeight allows
+    public void AddItem(Item newItem, int amount = 1)
+    {
+        // return if adding goes overweight
+        if (GetCurrentWeight()+newItem.weight * amount > maxWeight) { return; }
+        foreach (InventorySlot slot in inventory) 
+        {
+            if (slot.GetItem() == newItem) 
+            {
+                slot.AddItem(amount);
+                return;
+            }
+        }
+        inventory.Add(new InventorySlot(newItem, amount));
+    }
+
+    //-/////////////////////////////////////////////////////////////////////
+    /// returns weight of all inventory slots
+    public int GetCurrentWeight() 
```

Comment:
Looping through the entire list of inventory slots every time we want to get the current weight is expensive.

Instead, I suggest creating `currentWeight` into its own variable.
`public float currentWeight { private get; public set }`
This should make currentWeight available to the public, but only settable from within the inventory.

In addition, we should `UpdateCurrentWeight(float)`. Negative numbers decrease the currentWeight, and positive numbers add to it. We can call this from methods like `AddItem()` or `RemoveItem()`